usl_pipeline/cloud_functions/main.py

```diff
@@ -563,6 +564,16 @@ def _process_wps_feature(feature: xarray.DataArray, var_config: dict) -> NDArray
     if "num_metgrid_levels" in feature.dims:
         feature = feature.isel(num_metgrid_levels=0)
 
+    # Monthly climatology var - extract to only the month of the file's datestamp
+    if "z-dimension0012" in feature.dims and snapshot_time:
+        snapshot_datetime = datetime.datetime.strptime(
+            snapshot_time.values[0].decode(), "%Y-%m-%d_%H:%M:%S"
+        )
+        feature = feature.isel({"z-dimension0012": snapshot_datetime.month - 1})
```

Comment:
What's the `- 1` in `snapshot_datetime.month - 1` here?

Reply:
we subtract one here to get the corresponding index. For example if the snapshot month is July (7), then the value from the z-dim we want will be index=6